code/modules/mob/living/simple_animal/hostile/megafauna/chaos_marine.dm

```diff
@@ -0,0 +1,510 @@
+			"<span class='userdanger'>You butcher [L], restoring your health!</span>")
+			if(!is_station_level(z) || client) //NPC monsters won't heal while on station
+				adjustHealth(-(L.maxHealth * 0.5))
+			L.gib()
+			if(prob(50))
+				telegraph()
+				say(pick("Blood for the blood god!", "Skulls for the skull throne!", "Blood! Blood!!", "Die! Die! Die!", "You will be a fine offering!", "Chaos!"))
+			return TRUE
+	weapon.melee_attack_chain(src, target)
+	return TRUE
+
+/mob/living/simple_animal/hostile/megafauna/chaos_marine/Move()
+	if(charging)
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/megafauna/chaos_marine/proc/adjustCMspeed()
+	if(health <= maxHealth*0.25)
+		speed = 4
+		dash_mod = 0.4
+		dash_num = 5
+		rapid_melee = 3.5
+	else if(health <= maxHealth*0.5)
+		speed = 4.8
+		dash_mod = 0.6
+		dash_num = 4
+		rapid_melee = 2.5
+	else if(health <= maxHealth*0.75)
+		speed = 5.4
+		dash_mod = 0.8
+		dash_num = initial(dash_num)
+		rapid_melee = 1.5
+	else if(health > maxHealth*0.75)
+		speed = initial(speed)
+		dash_mod = initial(dash_mod)
+		dash_num = initial(dash_num)
+		rapid_melee = initial(rapid_melee)
+
+/mob/living/simple_animal/hostile/megafauna/chaos_marine/do_attack_animation(atom/A, visual_effect_icon, obj/item/used_item, no_effect)
+	if(!used_item && !isturf(A))
+		used_item = weapon
+	..()
+
+/mob/living/simple_animal/hostile/megafauna/chaos_marine/OpenFire()
+	adjustCMspeed()
+
+	if(client)
+		switch(chosen_attack)
+			if(1)
+				blood_dash(target)
+			if(2)
+				runic_blast()
+			if(3)
+				blast()
+			if(4)
+				infernal_summon()
+			if(5)
+				teleport_b(target)
+			if(6)
+				rapid_fire()
+		return
+
+	Goto(target, move_to_delay, minimum_distance)
+	if(get_dist(src, target) >= 1 && dash_cooldown <= world.time && !charging)
+		blood_dash(target)
+	if(get_dist(src, target) > round(5*dash_mod) && teleport_cooldown <= world.time && !charging)
+		teleport_b(target)
+	if(get_dist(src, target) > round(4) && ranged_cooldown <= world.time && !charging)
+		if(health <= maxHealth*0.5)
+			rapid_fire()
+		else
+			blast()
+	if(get_dist(src, target) <= 3 && runic_blast_cooldown <= world.time && !charging)
+		runic_blast()
+	if(infernal_summon_cooldown <= world.time && !charging)
+		infernal_summon()
+
+/obj/projectile/bullet/chaos_bomb
+	name ="bolter bullet"
+	desc = "Oh god oh fu..."
+	icon_state = "bolter"
+	damage = 20
+
+/obj/projectile/bullet/chaos_bomb/on_hit(atom/target, blocked = FALSE)
+	..()
+	explosion(target, -1, 0, 0, 0, 0, flame_range = 2)
+	return BULLET_ACT_HIT
+
+/obj/projectile/bullet/chaos_bomb/blood
+	name = "blood bolt"
+	icon_state = "mini_leaper"
+	damage = 10
+	nondirectional_sprite = TRUE
+	impact_effect_type = /obj/effect/temp_visual/dir_setting/bloodsplatter
+
+/mob/living/simple_animal/hostile/megafauna/chaos_marine/ex_act(severity, target)
+	return //Resistant to explosions
+
+/mob/living/simple_animal/hostile/megafauna/chaos_marine/death(gibbed)
+	if(health > 0)
```

Comment:
why is this health > 0 check here?

Reply:
I actually have no idea, this one is copy-pasted from other megas.